instrumentation/rediscala-1.8/src/main/java/io/opentelemetry/auto/instrumentation/rediscala/RediscalaInstrumentation.java

```diff
@@ -83,10 +83,10 @@ public RediscalaInstrumentation() {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
     public static SpanWithScope onEnter(@Advice.Argument(0) final RedisCommand cmd) {
-      final Span span =
-          TRACER.spanBuilder(cmd.getClass().getName()).setSpanKind(CLIENT).startSpan();
+      String statement = DECORATE.spanNameForClass(cmd.getClass());
+      final Span span = TRACER.spanBuilder(statement).setSpanKind(CLIENT).startSpan();
       DECORATE.afterStart(span);
-      DECORATE.onStatement(span, cmd.getClass().getName());
+      DECORATE.onStatement(span, statement);
```

Comment:
Does not look right to me. Passing span name as statement?

Reply:
Old code passed `cmd.getClass().getName()` to `onStatement` method. Ok, looks good: using class name for `RedisCommand` subclass as query string. 

New code calls `DECORATE.spanNameForClass` and decides to pass this to `onStatement` method. Does not look good: using span name (which can change in future) as query string?

Reply:
from https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md:

> Span name should be set to low cardinality value representing the statement executed on the database

in this case the statement is already low cardinality, so it serves for both span name and statement.

i'm not clear on what you'd like to change here, can you propose a specific change?

Reply:
I propose to pass `cmd.getClass().getName()` to `DECORATE.onStatement` method instead of the result of the `DECOREATE.spanNameForClass()`.

Look at it this way: does `DECORATE.onStatement(span, DECOREATE.spanNameForClass(cmd))` make sense to you? Question is not in cardinality, question is in semantics. `onStatement` expects to receive the equivalent of database query. Span name can be changed at any moment and may be completely different from database query.

Reply:
> Span name can be changed at any moment

not without breaking our tests

why would we calculate the same value twice, when we can calculate it once and use it in two places?

Reply:
So you are Ok with `DECORATE.onStatement(span, DECOREATE.spanNameForClass(cmd))`? For me this code is currently passing _span name_ as a _db query_. It just happens atm that they have the same value, but this is accidental.

Technically this code is right now correct, so I will not object to this PR. But this "homonym" will bite us in the future, I am sure :)

Reply:
will your concern go away once we migrate to the new `DatabaseClientTracer` (where you've made the `spanName` / `statement` separation more clear)?

Reply:
Hm, in fact I believe yes :)